src/VisualStudio/IntegrationTest/TestUtilities/AutomationElementExtensions.cs

```diff
@@ -299,5 +331,38 @@ private static string GetNameForExceptionMessage(this IUIAutomationElement eleme
         {
             return element.CurrentAutomationId ?? element.CurrentName ?? "<unnamed>";
         }
+
+        private static void RetryIfNotAvailable<T>(Action<T> action, T state)
+        {
+            for (var i = 0; true; i++)
```

Comment:
>true [](start = 28, length = 4)

I'd rather you make this explicitly `i < AutomationRetryCount`, and throw an unreachable exception after the try/catch. It took me a minute of verifying exit conditions to actually be sure this wasn't an infinite loop, and that pattern will much easier to reason about your expectations.

Reply:
💭 The issue I encountered here is it doesn't change the way the exception filter is used, so technically such a change would increase the cyclomatic complexity of the method but not actually add any reachable paths.

Reply:
It might increase some complexity of the method, but imo would result in code that is more easily understandable and verifiable.

---
In reply to: [178400103](https://github.com/dotnet/roslyn/pull/25455#discussion_r178400103) [](ancestors = 178400103)

Reply:
➡️ I went with a documentation approach instead. Hopefully it's enough to make it reasonably clear.